waspc/data/Generator/templates/react-app/src/auth/pages/createAuthRequiredPage.jsx

```diff
@@ -4,28 +4,27 @@ import React from 'react'
 import { Redirect } from 'react-router-dom'
 import { useAuth } from 'wasp/client/auth'
 
+import { DefaultLoader } from '../../DefaultLoader'
 
 const createAuthRequiredPage = (Page) => {
   return (props) => {
-    const { data: user, isError, isSuccess, isLoading } = useAuth()
+    const { data: user, status, error } = useAuth();
 
-    if (isSuccess) {
-      if (user) {
-        return (
-          <Page {...props} user={user} />
-        )
-      } else {
-        return <Redirect to="{= onAuthFailedRedirectTo =}" />
-      }
-    } else if (isLoading) {
-      return <span>Loading...</span>
-    } else if (isError) {
-      return <span>An error ocurred. Please refresh the page.</span>
-    } else {
-      return <span>An unknown error ocurred. Please refresh the page.</span>
+    switch (status) {
+      case 'success':
+        if (user) {
+          return (
+            <Page {...props} user={user} />
+          );
+        } else {
+          return <Redirect to="{= onAuthFailedRedirectTo =}" />;
+        }
+      case 'loading':
+        return <DefaultLoader/>;
+    case 'error':
+      return <span>An error occurred. Please refresh the page. {error?.message}</span>
```

Comment:
I'd suggest using the same error compnent as we used in the `OAuthCallbackPage`.

This component is not extracted in a separate file, but I can do that now in a simple PR.

Reply:
Here's the PR extracting the component: https://github.com/wasp-lang/wasp/pull/1991

I'd recommend using `<MessageError>` instead of the `<span>` to keep the look consistent :)